packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -718,6 +749,44 @@ function commitRoot(root: FiberRoot, finishedWork: Fiber): void {
     legacyErrorBoundariesThatAlreadyFailed = null;
   }
   onCommit(root, earliestRemainingTimeAfterCommit);
+
+  if (enableInteractionTracking) {
+    __interactionsRef.current = prevInteractions;
+
+    let caughtError = null;
+    let subscriber;
+
+    try {
+      subscriber = __subscriberRef.current;
+      if (subscriber !== null && root.memoizedInteractions.size > 0) {
+        // Interaction threads are unique per root and expiration time.
+        // DO NOT CHANGE THIS VALUE without also updating other threadID values.
+        const threadID =
```

Comment:
If this is so fragile, can we move it into a function?

Reply:
Sure. I didn't want to add the overhead of a function call, but maybe such a small function would just get inlined.